steps/24_skpro_polars_support_integration/step.md

```diff
@@ -0,0 +1,410 @@
+
+* polars will return polars when `predict` is called
+* pandas will return pandas when `predict` is called
+
+The output of `_predict_*`  functions will return pandas Dataframes no matter the input passed in as `X`.
+
+Question 2.1: For underlying interfaces that currently do not support polar Dataframe inputs (eg. statsmodels), is the expectation that we will refactor code that converts the data container into something it accepts or will we use the `_tags` dict via `X_inner_mtype` and `y_inner_mtype` convention to specify to the user what types of data containers it accepts. So for `statsmodels` case, polars Dataframes will be prohibited.
+
+Question 2.2: Follow up to question 2.1, should we go through every implemented regression estimator right now to see which data containers it accepts, as every interface could be different. Dummy data should be able to suffice?
+
+
+## 3) Extending functionality of `skpro.regression` and description of proposed solution
+
+Lets consider the current inplementation of `predict_interval` seen inside `regression._base`. Assuming we follow section 8.2's implementation, the proposed implementation of a function like `predict_interval` could look as follows:
+
+```python
+	    def predict_interval(self, X=None, coverage=0.90):
+        """Compute/return interval predictions.
+
+        If coverage is iterable, multiple intervals will be calculated.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas/polars DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+        coverage : float or list of float of unique values, optional (default=0.90)
+           nominal coverage(s) of predictive interval(s)
+
+        Returns
+        -------
+        pred_int : pd.DataFrame or pl.DataFrame
+            For pd.DataFrame: Column has multi-index: first level is variable name from ``y`` in fit,
+            second level coverage fractions for which intervals were computed,
+            in the same order as in input `coverage`.
+            Third level is string "lower" or "upper", for lower/upper interval end.
+            Row index is equal to row index of ``X``.
+            Entries are lower/upper bounds of interval predictions,
+            for var in col index, at nominal coverage in second col index,
+            lower/upper depending on third col index, for the row index.
+            Upper/lower interval end are equivalent to
+            quantile predictions at alpha = 0.5 - c/2, 0.5 + c/2 for c in coverage.
+
+	    For pl.DataFrame: Column is in single level following similar convention to pd.DataFrame
```

Comment:
that would work - although in the `pd.DataFrame` convention the second level is float. Do we forsee any issues in coercing everything to string? This no longer allows, for instance, to distinguish variable names that are string and those that are integer. I suppose the variable name is only an issue in mixed input/output type, if same, then that would always be str.

Reply:
I don't think column names in polars are allowed to be anything but strings. I've tried setting 0 or 0.9 as a column name but I got an error saying `TypeError: Series name must be a string`